### DISCUSSION TITLE
Shorthand use case to defining variants

### DISCUSSION THREAD

**Discussion by kerimcet** [stitchesjs/stitches#796] [Help] [answered]
Is there any shorthand use case to variant definition?

```js
variants: {
  spaceY: {
    1: { 
      spaceY: "$1" 
    },
    5: { 
      spaceY: "$5" 
    },
    10: { 
      spaceY: "$10" 
    }
  }
}
```

I would like to use this instead of code above:

```js
variants: {
  spaceY: {
    1: "$1",
    5: "$5",
    10: "$10" 
  }
}
```

Also that might be a feature idea if its possible.

**Comment by lukemorales** [comment, marked as answer]
Your idea is very similar to this PR implementation: https://github.com/modulz/stitches/pull/721, they're discussing if this feature fits the plans for Stitches.

But your approach with the PR would become:
```ts
variants: {
  spaceY: (value: 1 | 5 | 10) => ({
    spaceY: value
  )}
}
```

or even better, if you want that to be dynamic with every space token:
```ts
variants: {
  spaceY: (value: Stitches.ScaleValue<'space'>) => ({
    spaceY: value
  )}
}
```

then you would be able to use:
```jsx
<Component spaceY="$5" />
```

**Reply by kerimcet** [reply]
Thanks for the info! 😊
